RestaurantReservation.Db.Test/RepositoriesTest/EmployeeRepositoryTest.cs

```diff
@@ -0,0 +1,130 @@
+﻿using AutoFixture;
+using FluentAssertions;
+using Moq;
+using RestaurantReservation.Db.Entities;
+using RestaurantReservation.Db.Abstracts;
+using RestaurantReservation.Db.Enums;
+using RestaurantReservation.Db.Repositories;
+using RestaurantReservation.Db.Data;
+
+namespace RestaurantReservation.Db.Test.RepositoriesTest;
+public class EmployeeRepositoryTest
+{
+    private readonly Mock<RestaurantReservationDbContext> _mockDbContext;
+    private readonly Mock<IEmployeeRepository> _mockEmployeeRepository;
+    private readonly Fixture _fixture;
+    private readonly EmployeeRepository _employeeRepository;
+
+    public EmployeeRepositoryTest()
+    {
+        _mockEmployeeRepository = new Mock<IEmployeeRepository>();
+        _fixture = new Fixture();
+        _fixture.Behaviors.Remove(new ThrowingRecursionBehavior());
+        _fixture.Behaviors.Add(new OmitOnRecursionBehavior());
+        _mockDbContext = new Mock<RestaurantReservationDbContext>();
+    }
+
+     [Fact]
+    public async Task GetEmployees_ShouldReturnOkResult_WithListOfEmployees()
+    {
+        // Arrange
+        var employees = _fixture.CreateMany<Employee>(10).ToList();
+        _mockEmployeeRepository.Setup(repo => repo.GetAllAsync()).ReturnsAsync(employees);
+
+        // Act
+        var result = await _mockEmployeeRepository.Object.GetAllAsync();
+
+        // Assert
+        var okResult = result.Should().BeOfType<List<Employee>>().Subject;
+        okResult.Should().BeEquivalentTo(employees);
+    }
+
```

Comment:
Here, you are testing the mock library not the real database, if this is an integration test you have to insert to the database then get the results, you can use `InMemeoryDatabase`. Do it for all repo tests